application/ui/src/components/zoom/zoom.store.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { clampBetween } from '@geti/smart-tools/utils';
+import { create } from 'zustand';
+import { devtools } from 'zustand/middleware';
+
+import { ZoomState } from './types';
+import { getZoomState, ZOOM_STEP_DIVISOR } from './util';
+
+interface ZoomStore extends ZoomState {
+    setZoom: (zoom: Partial<ZoomState> | ((state: ZoomState) => Partial<ZoomState>)) => void;
+    fitToScreen: () => void;
+    onZoomChange: (factor: number, hasAnimation?: boolean) => void;
+    resetZoom: () => void;
+}
+
+const initialState: ZoomState = {
+    scale: 1.0,
+    maxZoomIn: 1,
+    hasAnimation: false,
+    translate: { x: 0, y: 0 },
+    initialCoordinates: { scale: 1.0, x: 0, y: 0 },
+};
+
+export const zoomStore = create<ZoomStore>()(
+    devtools(
```

Comment:
to use ReduxDevTools without redux. 

<img width="846" height="438" alt="Screenshot 2025-10-17 at 10 49 32" src="https://github.com/user-attachments/assets/459be8df-43eb-4ae4-a45b-7cbad484f742" />